### PR TITLE
Optimize merge selection process and increase disk space availability logic

### PR DESCRIPTION
**Pull request by caozj1011**
## Description
This PR mainly has the following upgrades:
1. Added the selection criteria for merging tasks in the space. When the availability of disk space is insufficient, TsFile containing mods file is preferentially selected.
2. Compaction adds the CompactionTaskType attribute to a task and compacts the task to `NORMAL` and `MOD_SETTLE`. `NORMAL` is the default value. When the disk size of the mod file exceeds 50M or the disk space availability is insufficient, the mods file sorting task is preferred. In this case, the task type is `MOD_SETTLE`. When a task is executed, the system determines the disk space availability. If the task type is `NORMAL` and the disk space availability is insufficient, the task is discarded from the priority queue to avoid unexpected errors.
3. Optimized the return value of the `selectInnerSpaceTask()` method. The previous return value was not convenient to wrap the CompactionTask object, so we made the logic of this method more pure, the purpose is to return a list of tasks that can be executed, which can be put directly into the task queue or directly executed.
4. Remove the priority queue in the in-space selection step. The `CompactionTaskManager` priority queue already contains all the priority decisions, so the priority queue in the selection stage is meaningless.

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java [hunk 1]
```diff
@@ -366,6 +366,11 @@ public boolean checkValidAndSetMerging() {
       resetCompactionCandidateStatusForAllSourceFiles();
       return false;
     }
+    if (compactionTaskType == CompactionTaskType.NORMAL && !CompactionUtils.isDiskHaveSpace()) {
+      LOGGER.debug(
```
Comment (xingtanzjr) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java:370]: using INFO is OK

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java [hunk 2]
```diff
@@ -366,6 +366,11 @@ public boolean checkValidAndSetMerging() {
       resetCompactionCandidateStatusForAllSourceFiles();
       return false;
     }
+    if (compactionTaskType == CompactionTaskType.NORMAL && !CompactionUtils.isDiskHaveSpace()) {
```
Comment (xingtanzjr) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java:369]: Maybe we can make this judgement as a method of parent class to get rid of duplicated codes in InnerSpaceCompaction

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/utils/CompactionUtils.java [hunk 3]
```diff
@@ -466,4 +472,29 @@ public static void deleteSourceTsFileAndUpdateFileMetrics(
       FileMetrics.getInstance().deleteFile(fileSizes, seq, fileNames);
     }
   }
+
+  public static boolean isDiskHaveSpace() {
```
Comment (xingtanzjr) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/utils/CompactionUtils.java:476]: Maybe rename it to `isDiskHasSpace` or using another description ? 😄

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java (hunk 2 above)
Comment (caozj1011) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java:369]: good idea

Comment (caozj1011) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/CrossSpaceCompactionTask.java:369]: good idea

**Review by shuwenwei — APPROVED** [review]

**Review by shuwenwei — APPROVED** [review]

**Pull request merged**